### PR TITLE
(sink/cdc): fix stop table dead lock introduced in #8949

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

Issue Number: ref #8657 

### What is changed and how it works?

The dead lock is caused by `tableSinkWrapper.tableSinkMu`:
* one goroutine is calling `tableSinkWrapper.close`, which requires the lock;
* `tableSinkWrapper.close` can be blocked for a long while if the backend sink is hang;
* now another goroutine wants to close the processor, but it can't get the lock;
* so the processor can't be closed until the backend sink hang is over.

This should fix capture_suicide_while_balance_table failures.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/processor/sinkmanager/table_sink_worker.go [hunk 1]
```diff
@@ -180,7 +180,7 @@ func (w *sinkWorker) handleTask(ctx context.Context, task *sinkTask) (finalErr e
 			// events have been reported. Then we can continue the table
 			// at the checkpoint position.
 			case tablesink.SinkInternalError:
-				task.tableSink.clearTableSink()
+				task.tableSink.clearTableSink(false)
```
Comment (nongfushanquan) [inline cdc/processor/sinkmanager/table_sink_worker.go:183]: please add some comments here to describe  the meaning of  "false"

**Review by nongfushanquan — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

**Pull request merged**